content/blog/2020-12-22-pipeline-II-analysis/index.md

```diff
@@ -0,0 +1,150 @@
+into some tool-specific observation knowledge, the estimation of RecMII is able to be close to real 
+numbers given by the tool. 
+
+## Evaluation
+I extracted 4 kernels in [PolyBench](http://web.cse.ohio-state.edu/~pouchet.2/software/polybench/)  that 
+contain loop-carried dependencies. Memory accesses in PolyBench kernels are all affine accesses, which 
+simplifies our analysis on the memory access index. The baseline numbers are given by running the key 
+kernel code in Catapult HLS tool.
+
+
+| Benchmarks  | Catapult HLS | My LLVM Pass | 
+| ----        | -------------| ------------ |
+| jacobi-1d   | ResMII = 3, RecMII = 1  | ResMII = 3, RecMII = 1  | 
+| jacobi-2d   | ResMII = 5, RecMII = 1  | ResMII = 5, RecMII = 1  | 
+| heat-3d     | ResMII = 7, RecMII = 1  | ResMII = 7, RecMII = 1  | 
+| seidel-2d   | ResMII = 9, RecMII = 3  | ResMII = 9, RecMII = 2  | 
+
+
+The experiment results show that the estimation of ResMII is precise for all test cases. The RecMII given
+by our analysis pass is a good estimation for most benchmarks, except seidel-2d. Actually, for all benchmarks 
+shown except seidel-2d, there is no real loop-carried dependency, and for this reason the pass returns 
+RecMII=1. Seidel-2d has a memory RAW dependency, and the dependency distance cannot be identified by LLVM 
+loop dependency analysis pass. Therefore, for this benchmark, the pass computes the index difference between 
+the dependency load-store pair, which is 1, and adds the tool-specific additional schedule cycles to it, 
+which outputs 2. It turns out that the estimation is over ideal, and the real II given by Catapult HLS tool is 3.
+
+## Future work
+ - For more precise estimation of ResMII, I need to add more types of resource constraints, e.g. interface ports, 
+and compute resources.
+ - For more precise estimation of RecMII, a more accurate and robust loop-carried dependency analysis is 
+ required to attain the exact dependency distance. In order to tune the analysis pass towards an HLS tool, 
+ I need to figure out a more systematic way to integrate scheduling information from the tool. The integration
+ part can be flexible and portable which also embraces other simulation or synthesis tools. 
+ [Gem5-SALAM](https://github.com/TeCSAR-UNCC/gem5-SALAM/blob/master/GEM5README) is a 
+ recent work that leverages Gem5 hardware models to simulate system-scale accelerators. 
+
+## References:
+ -  [xilinx2015_2/sdsoc_doc manual](https://www.xilinx.com/support/documentation/sw_manuals/xilinx2015_2/sdsoc_doc/topics/calling-coding-guidelines/concept_pipelining_loop_unrolling.html#:~:text=A%20data%20dependence%20from%20an,called%20a%20loop%2Dcarried%20dependence.&text=In%20case%20of%20loop%20pipelining,operation%2C%20and%20the%20memory%20write)
+ - Seto, Kenshu. "Scalar Replacement with Polyhedral Model." IPSJ Transactions on System LSI Design Methodology 11 (2018): 46-56.

```

Comment:
FWIW, in the context of a blog post, I think it can be more helpful to just link to the relevant materials in context, right where you refer to them. That can help make it clear to the reader what you used these references for and why they should care about them. And unlike in a PDF, links are easy to make, so there's no need for a list of references centralized at the bottom! :smiley: